# HG changeset patch
# User Rik <rik@octave.org>
# Date 1549642709 28800
#      Fri Feb 08 08:18:29 2019 -0800
# Branch stable
# Node ID 1c3471ce9a46942ea9fbb701ba6eccaceae105c5
# Parent  736f26d34f79087e8b15327b54598e4fc8ef96b1
Fix display of column vector struct arrays in Variable Editor (bug #55666).

* variable-editor-model.cc (vector_struct_model::vector_struct_model):
In constructor, set the number of rows to display equal to the number
of data columns.

diff --git a/libgui/src/variable-editor-model.cc b/libgui/src/variable-editor-model.cc
--- a/libgui/src/variable-editor-model.cc
+++ b/libgui/src/variable-editor-model.cc
@@ -696,16 +696,17 @@ namespace octave
       // FIXME: should fill the window vertically and expand on scrolling
       // or resizing.  No extra cells horizontally.  New fields must be
       // added specially.
 
       m_data_rows = val.numel ();
       m_data_cols = val.nfields ();
 
       maybe_resize_rows (m_data_rows + 16);
+      m_display_cols = m_data_cols;
     }
 
     ~vector_struct_model (void) = default;
 
     // No copying!
 
     vector_struct_model (const vector_struct_model&) = delete;
 
